### PR TITLE
Give access to the issue project when trying to save a viewpoint

### PR DESCRIPTION
**Pull request by oliverguenther**
With attachment whitelisting, all calls trying to create an attachment will be subject to authorization calls. For adding a viewpoint in the importer, they were not receiving the correct project context as `viewpoint.issue.project` (which is the same as `viewpoint.issue.work_package.project` was not set during import.

### CODE REVIEW

File: modules/bim/lib/open_project/bim/bcf_xml/issue_reader.rb [hunk 1]
```diff
@@ -45,7 +45,7 @@ def synchronize_with_work_package
       # If there are already errors during the BCF issue creation, don't create or update the WP.
       return if issue.errors.any?
 
-      self.is_update = issue.work_package.present?
+      self.is_update = issue.work_package.persisted?
```
Comment (oliverguenther) [inline modules/bim/lib/open_project/bim/bcf_xml/issue_reader.rb:48]: Note that this became necessary as setting a project on the issue creates a work package implicitly, as it's a `has_one :project through: :work_package` relation

File: app/services/attachments/create_service.rb [hunk 2]
```diff
@@ -26,74 +26,76 @@
...
-  end
+    ##
+    # Create an attachment service bypassing the user-provided whitelist
+    # for internal purposes such as exporting data.
+    #
+    # @param user The user to call the service with
+    # @param whitelist A custom whitelist to validate with, or empty to disable validation
+    #
+    # Warning: When passing an empty whitelist, this results in no validations on the content type taking place.
+    def self.bypass_whitelist(user:, whitelist: [])
```
Comment (wielinde) [inline app/services/attachments/create_service.rb]: @oliverguenther Isn't that method here an exact copy of what is already declared in the `BaseService`? If so, shouldn't we remove it here?

Comment (oliverguenther) [reply app/services/attachments/create_service.rb]: It was meant to deleted, probably messed up the commit there

**Review by wielinde — CHANGES_REQUESTED** [review]
@oliverguenther Please find my inline comment. Apart from that this PR looks good to me. Thank you!

**Review by wielinde — APPROVED** [review]

**Pull request merged**